r/tests/testthat/test-Table.R

```diff
@@ -288,20 +288,20 @@ test_that("==.Table", {
 test_that("Table$Equals(check_metadata)", {
   tab1 <- Table$create(x = 1:2, y = c("a", "b"))
   tab2 <- Table$create(x = 1:2, y = c("a", "b"),
-    schema = tab1$schema$WithMetadata(list(some="metadata")))
+                       schema = tab1$schema$WithMetadata(list(some="metadata")))
 
   expect_is(tab1, "Table")
   expect_is(tab2, "Table")
   expect_false(tab1$schema$HasMetadata)
   expect_true(tab2$schema$HasMetadata)
   expect_match(tab2$schema$metadata, "some: metadata", fixed = TRUE)
 
-  expect_false(tab1 == tab2)
-  expect_false(tab1$Equals(tab2))
-  expect_true(tab1$Equals(tab2, check_metadata = FALSE))
+  expect_true(tab1 == tab2)
+  expect_true(tab1$Equals(tab2))
+  expect_false(tab1$Equals(tab2, check_metadata = TRUE))
 
-  expect_failure(expect_equal(tab1, tab2)) # expect_equal does check_metadata
-  expect_equivalent(tab1, tab2)            # expect_equivalent does not
+  expect_equal(tab1, tab2)       # expect_equal does not check_metadata
```

Comment:
I think that `expect_equal` still should check metadata, for consistency with the regular behavior of that assertion, and I'm not sure why your change is affecting this because the method that it calls is explicitly setting `check_metadata` and not relying on the `$Equals` method default value.

Reply:
I'm fine to revert that change. The idea was for the default notion of equality to exclude metadata comparison. If the semantics of `expect_equal` is generally to also check e.g. data.frame attributions then that makes sense to me

Reply:
Reverting, give me a minute